### DISCUSSION TITLE
How to access child components from parent component?

### DISCUSSION THREAD

**Discussion by yipeng-jiang** [zauberzeug/nicegui#615] [Q&A] [answered]
Hi,  I have created a parent component with several child components in a function like below:
```python
def create_parent_component():
    with ui.row() as parent_component:
        [ui.label(item) for item in list_of_strings]
    return parent_component

parent_component = create_parent_component()
```
I am struggling to find a way to access the values of child components like 
```python
[child_component.text for child_component in parent_component.some_method]
```
Any idea which method should I use for the parent_component or I have to return each one of child component from the function.

**Comment by falkoschindler** [comment, marked as answer]
You are looking for `default_slot.children`:

```py
[child_component.text for child_component in parent_component.default_slot.children]
```

This is because UI elements can have multiple slots, each with a number of children. By default, every element has a `default_slot`, which is usually the place to look for its children.

This API is not yet well documented, but we're working on it (#439).

**Reply by yipeng-jiang** [reply]
Thank you very much. This exactly answered my question. By the way, I saw there are similar attributes called `slots` and `parent_slot`. What are the different from `default_slot`?

**Reply by falkoschindler** [reply]
Great! 👍🏻

`parent_slot` is the slot of the parent element, where the child is living in. Using this slot's `parent` attribute one can get to the parent element. And `slots` is the dictionary containing the `default_slot` and all other slots (if there are any).

**Reply by yipeng-jiang** [reply]
I see. Thank u.
